tasks/zip.js

```diff
@@ -154,12 +156,22 @@ module.exports = function(grunt) {
             var fileDir = path.dirname(filepath);
 
             // Write out the content
-            grunt.verbose.writeln('Writing file: "' + filepath + '"');
             grunt.file.mkdir(fileDir);
-            fs.writeFileSync(filepath, content, {mode: fileObj.unixPermissions});
+            if ((fileObj.unixPermissions & 0xf000) === 0xa000) {
+              symlinks.push([filepath, content.toString('utf8')]);
+            } else {
+              grunt.verbose.writeln('Writing file: "' + filepath + '"');
+              fs.writeFileSync(filepath, content, {mode: fileObj.unixPermissions});
+            }
           }
         }
       });
+
+      // Process symlinks at the end, to ensure that the target files have been extracted
+      symlinks.forEach(function (symlink) {
```

Comment:
It's usually possible to use `ln -s` without needing the file to exist first. It will reference a bad target but it will still work. Does this really need to go outside of the loop?

Reply:
Good point, it isn't really necessary to defer creating the links. I only did this because I noticed this is what `unzip` does on macOS. I wonder if the developers of `unzip` had a good reason for making it behave this way.

Reply:
Ah, word. I've got no idea why either -- maybe it's a 1 node type at a time thing? Anyway, can we relocate the symlink creation into the loop itself? It'd be nice to see the `verbose.writeln` next to one another